code/datums/elements/weather_listener.dm

```diff
@@ -31,10 +31,10 @@
 	. = ..()
 	UnregisterSignal(source, list(COMSIG_MOVABLE_Z_CHANGED, COMSIG_MOB_LOGOUT))
 
-/datum/element/weather_listener/proc/handle_z_level_change(datum/source, old_z, new_z)
+/datum/element/weather_listener/proc/handle_z_level_change(datum/source, turf/old_loc, turf/new_loc)
 	SIGNAL_HANDLER
 	var/list/fitting_z_levels = SSmapping.levels_by_trait(weather_trait)
-	if(!(new_z in fitting_z_levels))
+	if(!(new_loc.z in fitting_z_levels))
 		return
 	var/datum/component/our_comp = source.AddComponent(/datum/component/area_sound_manager, playlist, list(), COMSIG_MOB_LOGOUT, fitting_z_levels)
```

Comment:
![image](https://user-images.githubusercontent.com/6209658/190537790-2d003af6-d268-4639-b04a-6c5efc189c38.png)
does this work then?

Reply:
DAMN IT